### PR TITLE
Clear out dead code in permissions branch

### PR DESCRIPTION
**Pull request by EMcQ-BES**
### Changes:

- Removing code that is no longer used
- Move a few functions into CRUDHandler format

### CODE REVIEW

File: packages/meditrak-server/src/routes/index.js [hunk 1]
```diff
@@ -181,13 +173,13 @@ export default {
   importUsers: catchAsyncErrors(importUsers),
   importOptionSets: catchAsyncErrors(importOptionSets),
   postChanges: catchAsyncErrors(postChanges),
-  pruneChanges: catchAsyncErrors(pruneChanges),
+  pruneChanges: catchAsyncErrors(pruneChanges), // TODO: Legacy endpoint?
```
Comment (edmofro) [inline packages/meditrak-server/src/routes/index.js:176]: Yep - safe to delete

File: packages/meditrak-server/src/routes/optionSets/EditOptionSets.js [hunk 2]
```diff
@@ -15,6 +15,10 @@ export class EditOptionSets extends EditHandler {
   }
 
   async editRecord() {
-    await this.updateRecord();
+    const optionSet = await this.models.optionSet.findById(this.recordId);
+    const originalData = await optionSet.getData();
+    const updatedModel = { ...originalData, ...this.updatedFields };
+
+    return this.models.optionSet.updateById(this.recordId, updatedModel);
```
Comment (edmofro) [inline packages/meditrak-server/src/routes/optionSets/EditOptionSets.js:22]: What's the benefit of this specific handling?

File: packages/meditrak-server/src/dataAccessors/index.js [hunk 3]
```diff
@@ -4,10 +4,6 @@
  */
```
Comment (edmofro) [inline packages/meditrak-server/src/dataAccessors/index.js:4]: Surprised there are still so many left! Where is e.g. `findSurveyScreenComponentsInSurvey` used?

**Review by edmofro — COMMENTED** [review]
Will need another round of regression testing after this...sorry @Klaus-S 😞

File: packages/meditrak-server/src/routes/optionSets/EditOptionSets.js (hunk 2 above)
Comment (EMcQ-BES) [reply packages/meditrak-server/src/routes/optionSets/EditOptionSets.js:22]: I thought it was odd, but it was one of the few endpoints with a specific handler in the dataAccessors folders so I thought there must have been a reason.

If you also don't understand/remember why it was done this way then maybe I'll just get rid of it.

**Review by edmofro — APPROVED** [review]

File: packages/meditrak-server/src/routes/optionSets/EditOptionSets.js (hunk 2 above)
Comment (edmofro) [reply packages/meditrak-server/src/routes/optionSets/EditOptionSets.js]: The original commit was in [this PR](https://github.com/beyondessential/meditrak-server/pull/249/files), where there was an obvious reason for the `editOption` to be handled specifically, but less obvious for `editOptionSet`. I'd be keen to drop it and then highlight it as an area for retesting.

**Pull request merged**